File: sortingSentence-1859.js
/**
 * @param {string} s
 * @return {string}
 */
var sortSentence = function(s) {
    // console.log(s.split(' '));
    let arr = s.split(' ');
    // console.log(arr[0].slice(-1));
    let map = {};
    for(i of arr) {
        map[i.slice(-1)] = i.slice(0, i.length-1);
    }
    // console.log(map);
    // console.log(Object.values(map).join(' '));
    return Object.values(map).join(' ')
};

File: sortingSentence-1859.py
class Solution(object):
    def sortSentence(self, s):
        """
        :type s: str
        :rtype: str
        """
        myDict=dict()
        s=s.split(' ')
        myLine=''
        for word in s:
            myDict[word[-1]]=word[:-1]
        for i in range(len(s)-1):
            myLine=myLine+str(myDict[str(i+1)])+" "
        myLine=myLine+str(myDict[str(len(s))])
        return myLine
